FF1Lib/Assembly/Symbols.cs

```diff
@@ -0,0 +1,2649 @@
+				var labels = fieldsList
+					.Where(field => ((BA)field.GetValue(null)).Bank == bank)
+					.ToDictionary(field => field.Name, field => (BA)field.GetValue(null));
+				return labels.OrderBy(entry => entry.Value.Addr).ToList();
+			}
+		}
+
+		/// <summary>
+		/// Provide dictionary versions of the contents of Symbols.Labels, Symbols.Constants, and Symbols.Variables.
+		/// </summary>
+		public static class AsDictionaries
+		{
+			private static Dictionary<string, TValue> _staticFieldsDictionaryForType<TValue>(Type containingType)
+			{
+				IList<FieldInfo> list = containingType.GetFields();
+				return list.ToDictionary(field => field.Name, field => (TValue) field.GetValue(null));
+			}
+
+			private static void _setStaticFieldsInType<TValue>(Type containingType, Dictionary<string, TValue> dictionary)
+			{
+				dictionary.ToList().ForEach(entry => containingType.GetField(entry.Key).SetValue(null, entry.Value));
+			}
+
+			public static Dictionary<string, BA> Labels
+			{
+				get => _staticFieldsDictionaryForType<BA>(typeof(Symbols.Labels));
+				set => _setStaticFieldsInType(typeof(Symbols.Labels), value);
+			}
+
+			public static Dictionary<string, int> Constants
+			{
+				get => _staticFieldsDictionaryForType<int>(typeof(Symbols.Constants));
+				set => _setStaticFieldsInType(typeof(Symbols.Constants), value);
+			}
+
+			public static Dictionary<string, int> Variables
+			{
+				get => _staticFieldsDictionaryForType<int>(typeof(Symbols.Variables));
+				set => _setStaticFieldsInType(typeof(Symbols.Constants), value);
+			}
+
+			public static Dictionary<string, int> VariablesAndConstants
+			{
+				get
+				{
+					var both = new Dictionary<string, int>();
+
+					// this is apparently a good and fast way to combine dictionaries, even if it looks a little stupid.
+					AsDictionaries.Constants.ToList().ForEach(x => both.Add(x.Key, x.Value));
+					AsDictionaries.Variables.ToList().ForEach(x => both.Add(x.Key, x.Value));
+
+					return both;
+				}
+			}
+
+			/// <summary>
+			/// Labels mapped to their jump locations, dropping bank information.
+			/// </summary>
+			public static Dictionary<string, int> LabelsWithRunAddresses
+			{
+				get
+				{
+					return AsDictionaries.Labels.ToDictionary(item => item.Key, item => item.Value.Addr);
+				}
+			}
+
+			/// <summary>
+			/// All symbols. NOTE: Label values are transformed into ints using BA.addr.
+			/// </summary>
+			public static Dictionary<string, int> All
+			{
+				get
+				{
+					var everything = new Dictionary<string, int>();
+
+					// this is apparently a good and fast way to combine dictionaries, even if it looks a little stupid.
+					AsDictionaries.Constants.ToList().ForEach(x => everything.Add(x.Key, x.Value));
+					AsDictionaries.Variables.ToList().ForEach(x => everything.Add(x.Key, x.Value));
+					AsDictionaries.Labels.ToList().ForEach(x => everything.Add(x.Key, x.Value.Addr));
+
+					return everything;
+				}
+			}
+
+		}
+
+		public static class Labels
```

Comment:
Wow, awesome.  I can see this being really useful even if you're not assembling.